phongo_compat.h

```diff
@@ -159,29 +159,68 @@
+		zval zchild;                                      \
+		php_phongo_new_int64(&zchild, (_value) TSRMLS_CC); \
+		add_next_index_zval((_zv), &zchild);             \
+	} else {                                              \
+		add_next_index_long((_zv), (_value));             \
 	}
-#define ADD_ASSOC_INT64(zval, key, value)                                                                                             \
-	if ((value) > INT32_MAX || (value) < INT32_MIN) {                                                                                 \
-		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Integer overflow detected on your platform: %lld", (value)); \
-	} else {                                                                                                                          \
-		add_assoc_long(zval, key, (value));                                                                                           \
+#define ADD_ASSOC_INT64(_zv, _key, _value)                 \
+	if ((_value) > INT32_MAX || (_value) < INT32_MIN) {     \
+		zval zchild;                                      \
+		php_phongo_new_int64(&zchild, (_value) TSRMLS_CC); \
+		add_assoc_zval((_zv), (_key), &zchild);           \
+	} else {                                              \
+		add_assoc_long((_zv), (_key), (_value));           \
 	}
-#else
+#else /* PHP_VERSION_ID >= 70000 */
+#define ADD_INDEX_INT64(_zv, _index, _value)              \
+	if ((_value) > INT32_MAX || (_value) < INT32_MIN) {    \
+		zval* zchild = NULL;                             \
+		TSRMLS_FETCH();                                  \
+		MAKE_STD_ZVAL(zchild);                           \
+		php_phongo_new_int64(zchild, (_value) TSRMLS_CC); \
+		add_index_zval((_zv), (_index), zchild);         \
+	} else {                                             \
+		add_index_long((_zv), (_index), (_value));        \
+	}
+#define ADD_NEXT_INDEX_INT64(_zv, _value)                \
+	if ((_value) > INT32_MAX || (_value) < INT32_MIN) {    \
+		zval* zchild = NULL;                             \
+		TSRMLS_FETCH();                                  \
+		MAKE_STD_ZVAL(zchild);                           \
+		php_phongo_new_int64(zchild, (_value) TSRMLS_CC); \
+		add_next_index_zval((_zv), zchild);             \
+	} else {                                             \
+		add_next_index_long((_zv), (_value));            \
+	}
+#define ADD_ASSOC_INT64(_zv, _key, _value)                \
+	if ((_value) > INT32_MAX || (_value) < INT32_MIN) {    \
+		zval* zchild = NULL;                             \
+		TSRMLS_FETCH();                                  \
+		MAKE_STD_ZVAL(zchild);                           \
+		php_phongo_new_int64(zchild, (_value) TSRMLS_CC); \
+		add_assoc_zval((_zv), (_key), zchild);           \
+	} else {                                             \
+		add_assoc_long((_zv), (_key), (_value));          \
+	}
+#endif /* PHP_VERSION_ID >= 70000 */
+#else  /* SIZEOF_PHONGO_LONG == 8 */
```

Comment:
The comment here makes it look like this is the end of the "size_long = 8" section, which is not the case, as it's the end of the "size_long = 4" section.

Reply:
This is no different than `#else /* PHP_VERSION_ID >= 70000 */` used above. IIRC, we discussed this a while back in another PR and the conclusion was that we should stick to the original `#if` condition to match things up. Would you prefer I write this as `SIZEOF_PHONGO_LONG != 8 && SIZEOF_PHONGO_LONG != 4` to be most accurate?

On the same token, I should probably add `SIZEOF_PHONGO_LONG != 8` to the `#endif` two lines below.